libs/ledger/src/chain/main_chain.cpp

```diff
@@ -1363,15 +1363,19 @@ bool MainChain::DetermineHeaviestTip()
     // find the heaviest item in our tip selection
     auto it = std::max_element(
         tips_.begin(), tips_.end(), [](TipsMap::value_type const &a, TipsMap::value_type const &b) {
-          auto        a_weight{a.second.total_weight}, b_weight{b.second.total_weight};
+          auto        a_total_weight{a.second.total_weight}, b_total_weight{b.second.total_weight};
           auto const &a_hash{a.first}, &b_hash{b.first};
+          auto        a_weight{a.second.weight}, b_weight{b.second.weight};
 
-          return a_weight < b_weight || (a_weight == b_weight && a_hash < b_hash);
+          return a_total_weight < b_total_weight ||
```

Comment:
👍 Using weight as a tiebreak right? Since it's eqivalent to rank - def worth putting in a comment here to explain why this is important

Reply:
Sure.